### ISSUE TITLE
Not able to delete service offering from UI or API

### ISSUE THREAD

**Issue by rohityadavcloud** [apache/cloudstack#9475]
I was not able to delete a service offering, upon calling the deleteServiceOffering API, it would transition the offering to Inactive state -> same as with enable/disable UI button. I don't know if this has been like this historically or there was some behaviour change in recent releases. I could mark them as removed in the DB, then they won't show up in the offering UI.

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
API, Service offerings
~~~

##### CLOUDSTACK VERSION

~~~
4.19.1
~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by weizhouapache** [comment]
this looks related to #8821 

cc @vishesh92

**Comment by harikrishna-patnala** [comment]
@rohityadavcloud I don't find any API which marks the service offering's removed column. deleteServiceOffering API and updateServiceOffering (using state parameter) API are updating the column "State" with Active and Inactive values.

This was the behaviour even before also.

With the PR https://github.com/apache/cloudstack/pull/8821 we have now new filter for Active and Inactive offerings which can also lists all types of offerings

Created two offerings and then deleted both the offerings from UI. Following are the screenshots for Active and Inactive offerings

Active offerings
![Screenshot 2024-08-19 at 3 57 01 PM](https://github.com/user-attachments/assets/30a56bdd-b167-43d9-8b70-c632019f4535)

Inactive offerings
![Screenshot 2024-08-19 at 3 57 09 PM](https://github.com/user-attachments/assets/9505702d-8d0d-4f0b-911b-9bb776ab8b68)

After enabled one of the offerings, it came back to Active list
![Screenshot 2024-08-19 at 3 57 29 PM](https://github.com/user-attachments/assets/47feefd6-d3d9-40ee-9284-639dfc5b544a)

Please check and let me know if anything else I'm missing here, otherwise we can close the ticket.

**Comment by DaanHoogland** [comment]
@rohityadavcloud can you expand on this? #8821 claims to fix this in 4.19.1, but you reported this in this version.

**Comment by li-liwen** [comment]
Hi, I am using CloudStack 4.20. I encountered the same issue. According to the author of #8821, the delete command in the API will no longer actually delete the offering. Instead, it will set the state to inactive. (https://github.com/apache/cloudstack/issues/8817#issuecomment-2014606121)

So is there a way to actually delete the service offering? (I guess making SQL operations would work, but I don't think it's appropriate for administrators to touch the SQL databse directly.)

**Comment by DaanHoogland** [comment]
@li-liwen If I read the changes in #8821 correctly it only changes the behaviour of the update<Disk/Service>Offering commands. So you should still be able to call the deleteServiceOffering API, for instance using cmk. 
cc @vishesh92 ??

**Comment by vishesh92** [comment]
it's been some time, but AFAIR it's kind of complicated and would require a lot of changes.

**Comment by DaanHoogland** [comment]
I think for now setting the `removed` field to `now()` is the best workaround.

**Comment by vishesh92** [comment]
> I think for now setting the `removed` field to `now()` is the best workaround.

Yes. This should work. But it can cause issues with operations for existing VMs. The user will have to ensure no VM is using the offering.

**Comment by li-liwen** [comment]
I got it. Thanks for helping. I would just leave them inactive for now.

**Comment by DaanHoogland** [comment]
@vishesh92 , we could implement the delete API in such a way that it does a check if th offering is used and if it is it 'only' disables it. If not used it would actually mark it as removed.

**Comment by rohityadavcloud** [comment]
Inactive as a workaround is okay - if historically there never was a way to delete offerings in the UI or deleteServiceOffering API didn't work, let's close this.

**Comment by DaanHoogland** [comment]
converting to discuss item as the resolution would only be the workaround.
